### PR TITLE
tests: ztest: base: remove main_deprecated file

### PR DESCRIPTION
**Pull request by shuai1x**
On intel_adsp_cavs25 platform, tests/ztest/base failed.
Failed reason: RunID mismatch
And the cause of the failure was found.
Because when move to new ztest API, the file was forgotten to delete.
So I delete it to solve this bug.

Signed-off-by: Zhao Shuai <shuai1x.zhao@intel.com>

### CODE REVIEW

**Review by aaronemassey — DISMISSED** [review]

**Comment by aaronemassey** [comment]
Please get a +1 from someone more familiar with the legacy API than me before merging.

**Comment by shuai1x** [comment]
> Please get a +1 from someone more familiar with the legacy API than me before merging.

Ok. I will.

**Pull request closed without merging**
